editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-helpers.ts

```diff
@@ -238,3 +239,41 @@ export function getInsertionPathForReparentTarget(
   }
   return conditionalClauseInsertionPath(newParent, clause, 'replace')
 }
+
+function areElementsInstancesOfTheSameComponent(
+  firstInstance: ElementInstanceMetadata | null,
+  secondInstance: ElementInstanceMetadata | null,
+): boolean {
+  if (
+    firstInstance == null ||
+    secondInstance == null ||
+    isLeft(firstInstance.element) ||
+    isLeft(secondInstance.element) ||
+    firstInstance.element.value.type !== 'JSX_ELEMENT' ||
+    secondInstance.element.value.type !== 'JSX_ELEMENT'
```

Comment:
```suggestion
    !isJSXElement(firstInstance.element.value) ||
    !isJSXElement(secondInstance.element.value)
```

Reply:
added it! [688b0dc](https://github.com/concrete-utopia/utopia/pull/3829/commits/688b0dcb43afe66e17d613faa358cadb0b594b13)